lib/active_admin/base_controller/authorization.rb

```diff
@@ -1,8 +1,13 @@
 module ActiveAdmin
   class BaseController < ::InheritedResources::Base
     module Authorization
-      include MethodOrProcHelper
       extend ActiveSupport::Concern
+      include MethodOrProcHelper
+
+      # prevent mixing of helper methods with controller action methods
+      MethodOrProcHelper.public_instance_methods(false).each do |method_name|
+        private method_name
+      end
 
```

Comment:
:scream: Why is MethodOrProcHelper included here @deivid-rodriguez ?  Oh yes, see dispatch_active_admin_access_denied in bb95d3d

Reply:
So if we use the alternative you proposed in #5084:

```diff
- call_method_or_exec_proc active_admin_namespace.on_unauthorized_access, exception
+ instance_exec(self, exception, &active_admin_namespace.on_unauthorized_access.to_proc)
```

Then we will no longer need this? If that's the case, let's re-add it to #5084 then, you convinced me :)

Reply:
Should I wait for #5084 to be merged, then rebase this one?

Reply:
@deivid-rodriguez  Unfortunately it is not so simple.  Since `MethodOrProcHelper` was included in `ResourceController` by #1817 `render_in_context()` has been used in two additional pull requests: #2089 (scope_to if/unless) and #3503 (batch_action options).  I have no problem with render_in_context(), but what I don’t like is that multiple generic instance_exec++ utility methods have gathered in a poorly cohesive module that is included in multiple places.  These two other references could be fixed with an explicit `MethodOrProcHelper.render_in_context` or by using `StringSymbolOrProcSetting.new(...).value`  Thoughts?

Reply:
I see. An explicit `MethodOrProcHelper` receiver so we don't need to include the module sounds good, but you maybe need to `extend self` on the module in order to be able to make that call?

The other alternative sounds neat too! I pretty much trust your judgement and I apologize for being a pain in the ass sometimes... :)

Reply:
@chumakoff #5176 has been merged, you can go ahead and rebase now.

Reply:
@varyonic Done. Added something new to `resource_controller/action_builder.rb`